content/docs/user-guide/networking/multiplayer/split.md

```diff
@@ -0,0 +1,207 @@
+        NAMESPACE Gem
+        FILES_CMAKE
+            multiplayersample_autogen_files.cmake
+            multiplayersample_files.cmake
+            ${pal_dir}/multiplayersample_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+        BUILD_DEPENDENCIES
+            PUBLIC
+                Gem::DebugDraw
+                Gem::PhysX
+                Gem::Multiplayer
+            PRIVATE
+                Gem::Multiplayer.Unified.Static
+                Gem::PhysX.Static
+                Gem::DebugDraw.Static
+                Gem::ImGui.Static
+        AUTOGEN_RULES
+            *.AutoComponent.xml,AutoComponent_Header.jinja,$path/$fileprefix.AutoComponent.h
+            *.AutoComponent.xml,AutoComponent_Source.jinja,$path/$fileprefix.AutoComponent.cpp
+            *.AutoComponent.xml,AutoComponentTypes_Header.jinja,$path/AutoComponentTypes.h
+            *.AutoComponent.xml,AutoComponentTypes_Source.jinja,$path/AutoComponentTypes.cpp
+    )
+```
+
+Meanwhile, Multiplayer_ScriptCanvas only requires core datatypes so it only uses Multiplayer.Common
+
+```cmake
+    ly_add_target(
+        NAME ${gem_name}.Static STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            scriptcanvas_multiplayer_files.cmake
+            scriptcanvas_autogen_files.cmake
+        BUILD_DEPENDENCIES
+            PUBLIC
+                Gem::ScriptCanvas
+            PRIVATE
+                Gem::Multiplayer.Common.Static
+    )
+```
+### Conditional Compilation
+
+MultiplayerComponents are subject to conditional compilation. This is done using the macros `AZ_TRAIT_CLIENT` and `AZ_TRAIT_SERVER`. Client specific logic should be wrapped in the former while Server specific logic should be wrapped in the latter. The motivation for this approach is to allow target specific logic in MultiplayerComponents without requiring target specific files (i.e. a ServerComponent and ClientComponent with or without a BaseComponent).
+
+In the Multiplayer Gem's cmake, observe that each target enables or disables these traits based on the target. In example, Server enables `AZ_TRAIT_SERVER` while disabling `AZ_TRAIT_CLIENT`. Usage of these targets will bring the macro definitions with them.
+
+{{< note >}}
+The following CMake example is abbreviated
+{{< /note >}}
+
+```cmake
+    ly_add_target(
+        NAME Multiplayer.Client.Static STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            multiplayer_split_files.cmake
+        COMPILE_DEFINITIONS
+            PUBLIC
+                AZ_TRAIT_CLIENT=1
+                AZ_TRAIT_SERVER=0
+        BUILD_DEPENDENCIES
+            PUBLIC
+                AZ::AzCore
+                AZ::AzFramework
+                AZ::AzNetworking
+                Gem::Multiplayer.Common.Static
+    )
+
+    ly_add_target(
+        NAME Multiplayer.Server.Static STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            multiplayer_split_files.cmake
+        COMPILE_DEFINITIONS
+            PUBLIC
+                AZ_TRAIT_CLIENT=0
```

Comment:
What happens in .Unified? Should user be setting both AZ_TRAIT_CLIENT and AZ_TRAIT_SERVER to 1?

Reply:
Or, in this case it would be .Common target?

Reply:
That's correct, do you think it's worth adding the cmake for it?

Reply:
+1 to adding an example for the unified (looks like it's already been added below)